core/src/main/scala/org/apache/spark/metrics/sink/StatsdReporter.scala

```diff
@@ -65,9 +66,11 @@ private[spark] class StatsdReporter(
       meters: SortedMap[String, Meter],
       timers: SortedMap[String, Timer]): Unit =
     Try(new DatagramSocket) match {
-      case Failure(ioe: IOException) => logWarning("StatsD datagram socket construction failed",
-        NetUtils.wrapException(host, port, NetUtils.getHostname(), 0, ioe))
-      case Failure(e) => logWarning("StatsD datagram socket construction failed", e)
+      case Failure(ioe: IOException) =>
+        logWarning(log"StatsD datagram socket construction failed " +
+          log"${MDC(ERROR, NetUtils.wrapException(host, port, NetUtils.getHostname(), 0, ioe))}")
```

Comment:
I don't think we need to change this one. We just need to focus on the main message